### PR TITLE
[PORT] Secret Gateways: Config loaded Away Missions + Anti-observing Z level traits

### PR DESCRIPTION
**Pull request by Bibby0110**
# Document the changes in your pull request

Ports https://github.com/tgstation/tgstation/pull/61719

Adds secret gateways, these away missions are map files loaded from a config folder, instead of being files in the repo.
The chance of an away mission being a secret one is config dependant.
Secret away missions have a secret z level trait applied to them, making them not visible by ghosts (other than admin ghosts) other than looking through another player's eyes (you dont get xray like you normally do) with the Observe verb like in the image below.
image
Until there are maps added to the folder and someone sets the config up, no secret away missions will show up on the servers.
Adds a map report verb for viewing map reports
image
If the map has errors in it, it doesn't load and it creates a map report
image

In addition you no longer squeak if you're flying or have no gravity
Squeaks happen when projectiles hit stuff

# Wiki Documentation

# Changelog

:cl: Fikou, ported by TheGamerDK
rscadd: Secret Gateways can now be a thing enabled by config. They are not on the repository, instead, loaded from a config folder.
rscadd: Adds the secret and anti-phasing z level traits used by Secret Gateways, secret prevents observers from looking at the z level in any way other than observing through another player's eyes
experimental: admin loaded away missions can be secret as well
/:cl:

### CODE REVIEW

**Comment by neeshacark** [comment]
Yessss finally I've been waiting on this forever

**Comment by adamsong** [comment]
What happens if someone tries to observe an admin ghost on a secret z level? And it is admin ghosts, not just ghosts of an admin right?

File: code/datums/components/squeak.dm [hunk 1]
```diff
@@ -18,7 +18,7 @@
 		return COMPONENT_INCOMPATIBLE
 	RegisterSignal(parent, list(COMSIG_ATOM_ENTERED, COMSIG_ATOM_BLOB_ACT, COMSIG_ATOM_HULK_ATTACK, COMSIG_PARENT_ATTACKBY), .proc/play_squeak)
 	if(ismovable(parent))
-		RegisterSignal(parent, list(COMSIG_MOVABLE_BUMP, COMSIG_MOVABLE_IMPACT), .proc/play_squeak)
+		RegisterSignal(parent, list(COMSIG_MOVABLE_BUMP, COMSIG_MOVABLE_IMPACT, COMSIG_PROJECTILE_BEFORE_FIRE), .proc/play_squeak)
```
Comment (adamsong) [inline code/datums/components/squeak.dm:21]: Undocumented change

File: code/datums/components/squeak.dm [hunk 2]
```diff
@@ -60,11 +60,8 @@
 		var/obj/item/I = AM
 		if(I.item_flags & ABSTRACT)
 			return
-		else if(istype(AM, /obj/item/projectile))
-			var/obj/item/projectile/P = AM
-			if(P.original != parent)
-				return
-	if(istype(AM, /obj/effect/dummy/phased_mob)) //don't squeek if they're in a phased/jaunting container.
+
+	if(AM.movement_type & (FLYING|FLOATING) || !AM.has_gravity())
```
Comment (adamsong) [inline code/datums/components/squeak.dm:64]: Undocumented change

**Review by adamsong — DISMISSED** [review]

**Comment by JamieD1** [comment]
Memory Concerns will test later

**Comment by Bibby0110** [comment]
> Memory Concerns will test later

this doesn't do anything at present

**Review by JamieD1 — APPROVED** [review]

**Review by JamieD1 — APPROVED** [review]

**Comment by JamieD1** [comment]
Art not required, mapping helper non player seen.

**Pull request merged**